src/main/java/org/wildfly/security/password/impl/OneTimePasswordAlgorithmParametersSpiImpl.java

```diff
@@ -54,7 +54,7 @@ protected void engineEncode(final ASN1Encoder encoder, final OneTimePasswordAlgo
     protected OneTimePasswordAlgorithmSpec engineDecode(final ASN1Decoder decoder) {
         decoder.startSequence();
         final String algorithm = decoder.decodeOctetStringAsString();
-        final byte[] seed = decoder.decodeOctetString();
+        final String seed = decoder.decodeOctetStringAsString();
```

Comment:
@fjuma Done

I'm sorry, my response was very fast before, without thinking about. I think as soon as we are using the same charset to encode/decode we are safe. Before we were using UTF-8 for both, but I agree xxxxIA5Strings are more suitable for seed in this case. In this case, I have updated `AbstractAlgorithmParametersSpiImplTest` as well since now only ASCII characters are valid characters for the algorithm parameters implementation.